File: backend/src/main/java/net/ugurkartal/backend/models/Purchase.java
package net.ugurkartal.backend.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "purchases")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Purchase {
    @Id
    private String id;
    @DBRef
    private Product product;
    @DBRef
    private Supplier supplier;
    private double purchasePrice;
    private int quantity;
    private double totalPrice;
    private LocalDateTime purchaseDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private boolean isActive;
}

File: frontend/src/types/Purchase.ts
export type Purchase = {
    id: string,
    productId: string,
    productName: string,
    supplierId: string,
    supplierName: string,
    purchasePrice: number,
    quantity: number,
    totalPrice: number,
    purchaseDate: Date,
    createdAt: Date,
    updatedAt: Date
    active: boolean
}